app/models/shelter.rb

```diff
@@ -16,6 +16,14 @@ def self.order_by_number_of_pets
       .order("pets_count DESC")
   end
 
+  def self.reverse_alphabetical
+    find_by_sql "SELECT DISTINCT ON (name) * FROM Shelters ORDER BY name DESC"
+  end
+
+  def self.with_applicants
+    joins(pets: {application_pets: :application}).select(:name).distinct.pluck(:name)
```

Comment:
Nice job using a `.joins`. I am also wondering if you could `.joins(pets: :applications) but you might then have to include  a `.where` .